src/plugins/rust/rust-parser-generator-trait.js

```diff
@@ -173,7 +187,7 @@ const RustParserGeneratorTrait = {
       return {action: '', types};
     }
 
-    const typesRe = /\s*\|([^|]*)\|\s*->\s*(\w+);/g;
+    const typesRe = /\s*\|([^|]*)\|\s*->\s*(\w[\w<>, ']*);/g;
```

Comment:
is this to support type params?

Reply:
Yes, but I made mistake here. Correct variant is https://regex101.com/r/OtXlxb/1
Will fix it as well.

Reply:
done

Reply:
I test for this functionality on this example: https://gist.github.com/anru/1c0681620f041604cff36093b549a647#file-static-str-test-calc-rs-g and everything works well.